python/cudf/cudf/core/column/column.py

```diff
@@ -1356,7 +1356,7 @@ def as_column(arbitrary, nan_as_null=None, dtype=None, length=None):
             arbitrary = arbitrary.astype(dtype)
 
         if arb_dtype.kind == "M":
-            data = datetime.DatetimeColumn.from_numpy(arbitrary)
+            data = as_column(pa.array(arbitrary), nan_as_null=nan_as_null)
```

Comment:
Is this still zero copy going through pyarrow? Also concerned this is now not handling `NaT` values the same way we were previously.

Reply:
discussed offline and extracted common null handling logic to be able to be called from different places.